src/libs/fsdrv/lv_fs_memfs.c

```diff
@@ -0,0 +1,233 @@
+﻿/**
+ * @file lv_fs_memfs.c
+ *
+ * File System Interface driver for memory-mapped files
+ *
+ * This driver allows using a memory area as a file that can be read by normal file operations. It can
+ * be used, e.g., to store font files in slow flash memory, and load them into RAM on demand.
+ *
+ * You can enable it in lv_conf.h:
+ * 
+ * #define LV_USE_FS_MEMFS 1
+ * 
+ * The actual implementation uses the built-in cache mechanism of the file system interface.
+ *
+ * Since this is not an actual file system, file write and directories are not supported.
+ *
+ * The default drive letter is 'M', but this can be changed in lv_conf.h:
+ *
+ * #define LV_FS_MEMFS_LETTER 'M'
+ *
+ * To use it seamlessly with the file system interface a new extended path object has been introduced:
+ *
+ * lv_fs_path_ex_t mempath;
+ *
+ * This structure can be initialized with the helper function:
+ *
+ * lv_fs_make_path_ex(&mempath, (const uint8_t *) & my_mem_buffer, sizeof(my_mem_buffer));
+ *
+ * Then the "file" can be opened with:
+ *
+ * lv_fs_file_t file;
+ * lv_fs_res_t res = lv_fs_open(&file, (const char *) & mempath, LV_FS_MODE_RD);
+ *
+ * The path object can be used at any place where a file path is required, e.g.:
+ *
+ * lv_font_t* my_font = lv_font_load((const char *) & mempath);
+ *
+ */
+
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lvgl.h"
+#if LV_USE_FS_MEMFS != '\0'
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+ /**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+//NOTE: although these functions do not do anything, we need them for the file system interface to function correctly
+static void* fs_open(lv_fs_drv_t* drv, const char* path, lv_fs_mode_t mode);
+static lv_fs_res_t fs_close(lv_fs_drv_t * drv, void * file_p);
+static lv_fs_res_t fs_read(lv_fs_drv_t * drv, void * file_p, void * buf, uint32_t btr, uint32_t * br);
+static lv_fs_res_t fs_seek(lv_fs_drv_t * drv, void * file_p, uint32_t pos, lv_fs_whence_t whence);
+static lv_fs_res_t fs_tell(lv_fs_drv_t * drv, void * file_p, uint32_t * pos_p);
+#if 0
+static lv_fs_res_t fs_write(lv_fs_drv_t* drv, void* file_p, const void* buf, uint32_t btw, uint32_t* bw);
+static void * fs_dir_open(lv_fs_drv_t * drv, const char * path);
+static lv_fs_res_t fs_dir_read(lv_fs_drv_t * drv, void * dir_p, char * fn);
+static lv_fs_res_t fs_dir_close(lv_fs_drv_t * drv, void * dir_p);
+#endif
```

Comment:
Please remove these if these as they are not supported.

Reply:
Done